orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,218 @@
+
+        # Info attributes
+        self.update_info()
+        info_box = gui.widgetBox(self.controlArea, 'Info')
+        gui.label(info_box, self, 'Documents: %(n_documents)s')
+        gui.label(info_box, self, 'Tokens: %(n_tokens)s')
+        gui.label(info_box, self, 'Types: %(n_types)s')
+        gui.label(info_box, self, 'Matching: %(n_matching)s')
+        gui.rubber(self.controlArea)
+
+        # Auto-commit box
+        gui.auto_commit(self.controlArea, self, 'autocommit', 'Commit', 'Auto commit is on', callback=self.commit)
+
+        # Search
+        c_box = gui.widgetBox(self.mainArea, orientation="vertical")
+        self.input = gui.lineEdit(c_box, self, '',
+                                         orientation=Qt.Horizontal,
+                                         sizePolicy=QSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed),
+                                         label='Query:')
+        self.input.textChanged.connect(self.refresh_search)
+
+        # Concordances view
+        self.conc_view = QTableView()
+        self.model = ConcordanceModel()
+        self.conc_view.setModel(self.model)
+        self.conc_view.setSelectionMode(QTableView.SingleSelection)
+        self.conc_view.setSelectionBehavior(QTableView.SelectRows)
+        self.conc_view.horizontalHeader().hide()
+        self.conc_view.setShowGrid(False)
+        self.mainArea.layout().addWidget(self.conc_view)
+
+    def sizeHint(self):
+        return QSize(600, 400)
+
+    def set_data(self, data=None):
+        self.reset_widget()
+        self.corpus = data
+        if data is not None:
+            if not isinstance(data, Corpus):
+                self.corpus = Corpus.from_table(data.domain, data)
+        self.model.set_data(self.corpus)
+        self.update_info()
+        self.commit()
+
+    def refresh_search(self):
+        self.word = self.input.text()
+        self.model.set_word(self.word)
+        if self.corpus is not None and self.word:
+            self.update_info()
+        self.conc_view.resizeColumnToContents(1)
+        self.resize_columns()
+        self.conc_view.resizeRowsToContents()
+
+    def resize_columns(self):
+        col_width = (self.conc_view.width() - self.conc_view.columnWidth(1)) / 2
+        self.conc_view.setColumnWidth(0, col_width)
+        self.conc_view.setColumnWidth(2, col_width)
+
+    def resizeEvent(self, event):
+        super().resizeEvent(event)
+        self.resize_columns()
+
+    def reset_widget(self):
+        # Corpus
+        self.corpus = None
+        # Widgets
+        self.input.clear()
+        self.update_info()
+        # Models/vars
+        self.word = ""
+
+    def update_info(self):
+        if self.corpus is not None:
+            self.n_documents = len(self.corpus)
+            self.n_matching = '{}/{}'.format(self.model.matching_docs(), self.n_documents)
+            self.n_tokens = sum(map(len, self.corpus.tokens)) if self.corpus.has_tokens() else 'n/a'
```

Comment:
This is copied from Corpus Viewer and I think it should remain this way for consistency within the add-on.